client/js/is-sum-formula.js

```diff
@@ -0,0 +1,54 @@
+const getColAndRow = function(cellId) {
+  const col = cellId[0];
+  const row = cellId.slice(1);
+  return [col, row];
+}
+
+
+
+const isValidSumFormula = function(string, numCols, numRows) {
+  // string must be in form
+  // =SUM([One capital letter][0 or more digits]:[One capital letter][0 or more digits])
+  const re = /=SUM\(([A-Z]\d*):([A-Z]\d*)\)$/;
+  const result = string.match(re);
+
+  // if string is in valid format, get col letters and row numbers
+  if (result !== null) {
+    const startCellCol = getColAndRow(result[1])[0].charCodeAt(0) - 64;
```

Comment:
You can simplify this further by capturing the column and row values from the formula string directly, in the same regex.  Just add a couple more pairs of parentheses:

```
re = /=SUM\(([A-Z])(\d*):([A-Z])(\d*)\)$/;
```

Around `[A-Z]` and around `\d*`, separately.

The result looks like:

```
matches = '=SUM(A15:A20)'.match(re)
> ["=SUM(A15:A20)", "A", "15", "A", "20", index: 0, input: "=SUM(A15:A20)"]
```
Or if you have just a column specification (no row numbers):

```
matches = '=SUM(A:A)'.match(re)
> ["=SUM(A:A)", "A", "", "A", "", index: 0, input: "=SUM(A:A)"]
```

Then you don't need `getColAndRow()` at all.